keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -313,11 +315,13 @@ client-side validation, albeit one that is error-prone and not officially
 supported).
 
 Long-term, we want to favor using out-of-tree solutions for client-side
-validation, though this idea is still in its infancy.
+validation. The [kubeconform](https://github.com/yannh/kubeconform) project is an example of an out-of-tree solution that does this.
 
-The [kubeval](https://www.kubeval.com/) project is an example of an out-of-tree solution that does this, and
-we will look into expanding its support of open API to v3, and investigate
-whether it makes sense as a permanent solution to client-side validation.
+Although likely to never be a perfect representation of the validation that the
+server performs, adding better support for these out-of-tree solutions that
+better utilize the progress made server-side with regards to performance,
+openapi v3, etc is definitely an avenue of future work worth exploring by
```

Comment:
I'm ready to commit to reviewing designs for this purpose, but I'm not aware of something concrete at this moment.  There's nothing concrete and this is a statement that we want to help the client leverage the server, not facilitate an attempt to move server responsibility client-side, correct?

Reply:
cc @lavalamp 

for statement of future intent.

Reply:
You are correct that the future work is to have any client side validation tooling better leverage the information published by the server. 

The goal of this statement is not to commit API machinery to working on client side validation tooling but rather suggest that it would behoove those looking to use/build client side validation tools to understand and incorporate what the API server offers (via open API, etc) to most accurately reflect the validation that the server is doing (i.e. the source of truth).

Let me know if I should word it better to reflect this goal

Reply:
Good, I'm happy with that future intent.  I'd like to give daniel a couple days to have a comment prior to merge.

Reply:
Can we make this clearer or just remove it? If we have to say something, I suggest something more like: SIG API Machinery's effort will go to producing clear and complete (and improving over time) API spec documents instead of making client validation. We strongly recommend that third party integrators make use of such documents.

Reply:
Sounds good, added

Reply:
thanks, LGTM if @deads2k likes this wording!